x-pack/plugins/alerting/server/rules_client/rules_client.ts

```diff
@@ -644,42 +669,67 @@ export class RulesClient {
   }
 
   public async aggregate({
-    options: { fields, ...options } = {},
+    options: { fields, filter, ...options } = {},
   }: { options?: AggregateOptions } = {}): Promise<AggregateResult> {
-    // Replace this when saved objects supports aggregations https://github.com/elastic/kibana/pull/64002
-    const alertExecutionStatus = await Promise.all(
-      AlertExecutionStatusValues.map(async (status: string) => {
-        const { filter: authorizationFilter } = await this.authorization.getFindAuthorizationFilter(
-          AlertingAuthorizationEntity.Rule,
-          alertingAuthorizationFilterOpts
-        );
-        const filter = options.filter
-          ? `${options.filter} and alert.attributes.executionStatus.status:(${status})`
-          : `alert.attributes.executionStatus.status:(${status})`;
-        const { total } = await this.unsecuredSavedObjectsClient.find<RawAlert>({
-          ...options,
-          filter:
-            (authorizationFilter && filter
-              ? nodeBuilder.and([
-                  esKuery.fromKueryExpression(filter),
-                  authorizationFilter as KueryNode,
-                ])
-              : authorizationFilter) ?? filter,
-          page: 1,
-          perPage: 0,
-          type: 'alert',
-        });
+    const { filter: authorizationFilter } = await this.authorization.getFindAuthorizationFilter(
```

Comment:
In the `find` function, we have a try/catch around `getFindAuthorizationFilter` to appropriately audit any errors. Can we add that here too?

Reply:
Sure, thanks for the feedback!

Reply:
I created a new rule audit action `RuleAuditAction.AGGREGATE` with the same verbs and ECS action types of `FIND`. Let me know if this is suitable.